Projects/UOContent/Engines/Advanced Search/AdvancedSearchGump.cs

```diff
@@ -0,0 +1,888 @@
+            {
+                PushToWorkers(item);
+            }
+        }
+
+        foreach (var m in World.Mobiles.Values)
+        {
+            if (m.GetType().IsAssignableTo(type))
+            {
+                PushToWorkers(m);
+            }
+        }
+
+        // Block until everything is processed
+        for (var i = 0; i < _threadWorkers.Length; i++)
+        {
+            _threadWorkers[i].Sleep();
+        }
+
+        var ignoredEntities = new HashSet<IEntity>(ignoreQueue);
+
+        // Force the GC to collect the ignored entities
+        ignoreQueue.Clear();
+
+        var resultsList = new List<AdvancedSearchResult>(results.Count);
+        foreach (var result in results)
+        {
+            if (!ignoredEntities.Contains(result.Entity))
+            {
+                resultsList.Add(result);
+            }
+        }
+
+        SearchResults = resultsList.ToArray();
+
+        // Force the GC to collect the results
+        resultsList.Clear();
+
+        // Expand the gump
+        ShowExtension = true;
+
+        Resend(from);
+    }
+
+    private void SortList(Mobile from, int radioSwitch)
+    {
+        IComparer<AdvancedSearchResult> comparer;
+        switch (radioSwitch)
+        {
+            default:
+                {
+                    return;
+                }
+            case 0:
+                {
+                    SortByType = true;
+                    comparer = SortDescending
+                        ? AdvancedSearchResultTypeComparer.InstanceReverse
+                        : AdvancedSearchResultTypeComparer.Instance;
+                    break;
+                }
+            case 1:
+                {
+                    SortByName = true;
+                    comparer = SortDescending
+                        ? AdvancedSearchResultNameComparer.InstanceReverse
+                        : AdvancedSearchResultNameComparer.Instance;
+                    break;
+                }
+            case 2:
+                {
+                    SortByRange = true;
+                    comparer = new AdvancedSearchRangeComparer(from, SortDescending);
+                    break;
+                }
+            case 3:
+                {
+                    SortByMap = true;
+                    comparer = SortDescending
+                        ? AdvancedSearchResultMapComparer.InstanceReverse
+                        : AdvancedSearchResultMapComparer.Instance;
+                    break;
+                }
+            case 4:
+                {
+                    SortBySelected = true;
+                    comparer = SortDescending
+                        ? AdvancedSearchResultSelectedComparer.InstanceReverse
+                        : AdvancedSearchResultSelectedComparer.Instance;
+                    break;
+                }
+        }
+
+        if (SearchResults?.Length > 0)
+        {
+            Array.Sort(SearchResults, comparer);
```

Comment:
Technically we could sort in a background thread. Would we ever need to? Probably not?